civiclens/nlp/models.py

```diff
@@ -1,18 +1,90 @@
+from pathlib import Path
+
+from sentence_transformers import SentenceTransformer
 from transformers import (
+    AutoModelForSeq2SeqLM,
     AutoModelForSequenceClassification,
     AutoTokenizer,
+    T5ForConditionalGeneration,
+    T5Tokenizer,
     pipeline,
 )
 
 
-# sentiment models
-sentiment_model = "cardiffnlp/twitter-roberta-base-sentiment-latest"
+def model_path(model: str, tokenizer: bool, model_func) -> str:
+    """Returns the path to a saved huggingface model and downloads the model
+    if it hasn't been saved already
+
+    Args:
+        model (str): huggingface model id
+        tokenizer (bool): whether the item is a tokenizer
+        model_func (_type_): function used to download the item
+
+    Returns:
+        relative_path (str): relative path for accessing the model
+    """
+    # Define the relative path
+    if tokenizer:
+        relative_path = (
+            Path(__file__).resolve().parent
+            / "saved_models"
+            / model
+            / "tokenizer"
+        )
+    else:
+        relative_path = (
+            Path(__file__).resolve().parent / "saved_models" / model / "model"
+        )
+    # if model isn't saved, download it and return its path
+    if relative_path.exists():
+        return relative_path
+    else:
+        model_download = model_func(model)
+        model_download.save_pretrained(relative_path, from_pt=True)
+        return relative_path
+
+
+# title models
+title_model_path = model_path(
+    "google/flan-t5-base", False, T5ForConditionalGeneration.from_pretrained
+)
+title_model = T5ForConditionalGeneration.from_pretrained(title_model_path)
+title_tokenizer_path = model_path(
+    "google/flan-t5-base", True, T5Tokenizer.from_pretrained
+)
+title_tokenizer = T5Tokenizer.from_pretrained(title_tokenizer_path)
+
+# topic models
+sentence_transformer = SentenceTransformer("all-MiniLM-L6-v2")
```

Comment:
the SentenceTransformer class has a `cache_folder` option that lets you specify a directory to cache the model in, let's add that arg in so that model gets saved locally the first time this script is run!

Reply:
thanks for pointing that out!